post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java

```diff
@@ -131,25 +132,42 @@ protected void processCachedIds() {
                         processTopic(keyTopic, entity, ids, patientRepository::executeStoredProcForPatientIds);
                         break;
                     case INVESTIGATION:
-                        List<InvestigationResult> invData = processTopic(keyTopic, entity, ids, investigationRepository::executeStoredProcForPublicHealthCaseIds);
+                        List<InvestigationResult> invData = processTopic(keyTopic, entity, ids,
+                                investigationRepository::executeStoredProcForPublicHealthCaseIds);
                         ids.forEach(id -> {
                             if (idVals.containsKey(id)) {
-                                processId(id, idVals.get(id), investigationRepository::executeStoredProcForPageBuilder, Entity.CASE_ANSWERS);
+                                processId(id, idVals.get(id),
+                                        investigationRepository::executeStoredProcForPageBuilder, Entity.CASE_ANSWERS);
                                 idVals.remove(id);
                             }
                         });
-                        processTopic(keyTopic, Entity.F_PAGE_CASE, ids, investigationRepository::executeStoredProcForFPageCase);
+                        processTopic(keyTopic, Entity.F_PAGE_CASE, ids,
+                                investigationRepository::executeStoredProcForFPageCase);
                         datamartProcessor.process(invData);
+
+                        /* CNDIT-1584: Notifications has a dependency on Investigation and should be processed
+                        together to eliminate timing issues.*/
+                        Optional<Entry<String, List<Long>>> notificationIds = sortedEntries.stream()
+                                .filter(k -> k.getKey().contains(Entity.NOTIFICATIONS.getName()))
+                                .findFirst();
+                        if (notificationIds.isPresent()) {
+                            processTopic(
```

Comment:
Process notifications if there are any data in the idcache

Reply:
We're going to pass the investigation uids instead of notification uids to couple the updates. If there is no investigation update, primarily insert into INVESTIGATION dimension, the notification proc should not run the phc uid.

Reply:
Updated the notifications to directly work on investigation uids